addons/website_event/static/src/js/website_event_send_email.js

```diff
@@ -0,0 +1,94 @@
+/** @odoo-module **/
+
+import { rpc } from "@web/core/network/rpc";
+import { checkEmailValidity } from  "@mail/utils/common/format";
+import publicWidget from '@web/legacy/js/public/public_widget';
+
+publicWidget.registry.websiteEventSendEmail = publicWidget.Widget.extend({
+    selector: '.o_wevent_send_by_email_widget',
+    events: {
+        'click .o_wevent_js_send_email': '_onSendEmailClick',
+        'click .o_wevent_send_by_email_info .btn-close': '_onCloseInfoClick',
+    },
+
+    init() {
+        this.notification = this.bindService("notification");
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     * @param {Event} ev
+     */
+    _onSendEmailClick: async function (ev) {
+        ev.target.getElementsByTagName('i')[0].classList.remove('d-none');
+        ev.target.setAttribute('disabled', '');
+
+        const attendeesIds = JSON.parse(ev.target.dataset.attendeesIds) || [];
+        const eventId = parseInt(ev.target.dataset.eventId);
+        const ticketsHash = ev.target.dataset.ticketsHash;
+        const emails = document.querySelector('#o_wevent_send_by_email_input').value;
+        const emailInfo = checkEmailValidity(', ', emails)
+
+        if (emails === "" || emailInfo.invalidEmails.length) {
+            this._showError();
+        } else {
+            this._hideError();
+            await rpc(`/event/${ eventId }/my_tickets_by_email`, {
+                'emails': emails,
+                'registration_ids': attendeesIds,
+                'tickets_hash': ticketsHash,
+            });
+            this._showInfoMsg();
+        }
+        ev.target.removeAttribute('disabled');
+        ev.target.getElementsByTagName('i')[0].classList.add('d-none');
+    },
+
+    /**
+     * @private
+     * Show error message if email is invalid by adding the bootstrap class
+     * 'is-invalid' on the input field and its parent. The parent is the div that
+     * contains the input field and the error message and has the class 'has-validation'.
+     * For more information : https://getbootstrap.com/docs/5.3/forms/validation/#server-side
+     */
+    _showError: function () {
```

Comment:
_showEmailError ? Same for other method. The purpose is to give direct insight of what the method does. (this is an overall comment)